### PR TITLE
Feature/healthbar

### PR DESCRIPTION
**Pull request by PiterGroot**

### CODE REVIEW

File: Assets/Prefabs/HealthBar/Healthbar.prefab
Comment (lucasvdiepen) [inline Assets/Prefabs/HealthBar/Healthbar.prefab:1]: Change to `HealthBar` with a capital letter `B`

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace ProefExamen.Framework.UI.Hud
+{
+    /// <summary>
+    /// Class responsible for updating the player health bar
+    /// </summary>
+    public class HealthBar : MonoBehaviour
+    {
+        [SerializeField]
+        private Image _primaryBar, _secondaryBar;
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/Hud/HealthBar.cs]:
```suggestion
        [SerializeField]
        private Image _primaryBar;
        
        [SerializeField]
        private Image _secondaryBar;
```

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+using UnityEngine.UI;
+using UnityEngine;
+using DG.Tweening;
+
+using ProefExamen.Framework.Gameplay.PerformanceTracking;
+
+namespace ProefExamen.Framework.UI.Hud
+{
+    /// <summary>
+    /// Class responsible for updating the player health bar
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/Hud/HealthBar.cs]:
```suggestion
    /// Class responsible for updating the player health bar.
```

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        [SerializeField]
+        private Ease _ease = Ease.InOutBounce;
+
+        [SerializeField]
+        private RectTransform _rectTransform;
+        
+        private void Awake()
+        {
+            _performanceTracker.OnHealthChanged += UpdateHealthBar;
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/Hud/HealthBar.cs]: Please move event listeners to `OnEnable()` and `OnDisable()`

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+            _primaryBar.fillAmount = 1;
+            _secondaryBar.fillAmount = 1;
+        }
+
+        public void UpdateHealthBar(float currentHealth)
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/Hud/HealthBar.cs:47]: Mising summary

**Review by lucasvdiepen — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs (hunk 1 above)
Comment (PiterGroot) [reply Assets/Scripts/Framework/UI/Hud/HealthBar.cs]: grr

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        private void Awake()
+        {
+            _primaryBar.fillAmount = 1;
+            _secondaryBar.fillAmount = 1;
+        }
+
+        /// <summary>
+        /// Method responsible for updating health bar visual
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/Hud/HealthBar.cs]:
```suggestion
        /// Method responsible for updating health bar visual.
```

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        [SerializeField]
+        private Ease _ease = Ease.InOutBounce;
+
+        [SerializeField]
+        private RectTransform _rectTransform;
+
+        private void OnEnable()
+        {
+            _performanceTracker.OnHealthChanged += UpdateHealthBar;
+        }
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/Hud/HealthBar.cs]: Use lambda

File: Assets/Scripts/Framework/UI/Hud/HealthBar.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            DOTween.Sequence()
+                .AppendInterval(_secundaryDelay)
+                .Append(_secondaryBar.DOFillAmount(alpha, _secundaryResetDuration))
+                .SetEase(_ease);
+        }
+
+        private void OnDisable()
+        {
+            _performanceTracker.OnHealthChanged -= UpdateHealthBar;
+        }
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/Hud/HealthBar.cs]: Use lambda

**Review by lucasvdiepen — CHANGES_REQUESTED** [review]

**Review by lucasvdiepen — APPROVED** [review]

**Pull request merged**